Apps/Admin/Server/Services/AgentAccessService.cs

```diff
@@ -99,12 +99,12 @@ public async Task<AdminAgent> ProvisionAgentAccessAsync(AdminAgent agent)
             }
 
             List<UserRepresentation> getUserResponse = await this.keycloakAdminApi.GetUsersByUsernameAsync(user.Username, jwtModel.AccessToken).ConfigureAwait(true);
-            UserRepresentation createdUser = getUserResponse.First();
+            UserRepresentation createdUser = getUserResponse[0];
             await this.keycloakAdminApi.AddUserRolesAsync(createdUser.UserId.GetValueOrDefault(), roles, jwtModel.AccessToken).ConfigureAwait(true);
 
             string[] splitString = createdUser.Username.Split('@');
-            string createdUserName = splitString.First();
-            string createdIdentityProviderName = splitString.Last();
+            string createdUserName = splitString[0];
+            string createdIdentityProviderName = splitString[^1];
```

Comment:
This was a SonarCloud warning, because direct indexing of an IList has a performance benefit over using the IEnumerable First(), Last(), or ElementAt() methods, which have to do a type cast to IList before they can take advantage of indexing.

It's a non-issue with small collections, but disabling the rule seemed more trouble than going along with it.  (We're using a shared code quality profile for the entire organization.)

I do want to become more familiar with using ranges and the "from end" index operator in particular.

Reply:
I'd argue the most important check if the split resulted in 2 elements is missing. I'm with @BronzBierd here to favor readability over generic nonsense warnings: split, check there are exactly 2 elements, map with [0] and [1]; no 2 elements? a invalid op exception is in order.

Reply:
I agree that it would be nice to fail if the split didn't go as planned.  I'll try to fix that.

Reply:
I added a check to throw an exception if the string split leads to unexpected results.